Common/cpp/NativeModules/NativeReanimatedModule.cpp

```diff
@@ -230,23 +242,92 @@ void NativeReanimatedModule::unregisterEventHandler(
       [=] { eventHandlerRegistry_->unregisterEventHandler(id); });
 }
 
+inline std::string int_to_hex(int val) {
+  std::stringstream ss;
+  ss << '#' << std::setfill('0') << std::setw(6) << std::hex << (val | 0);
+  return ss.str();
+}
+
+#ifdef RCT_NEW_ARCH_ENABLED
+
+std::string NativeReanimatedModule::obtainPropFromShadowNode(
+    jsi::Runtime &rt,
+    const std::string &propName,
+    const ShadowNode::Shared &shadowNode) {
+  auto newestCloneOfShadowNode =
+      uiManager_->getNewestCloneOfShadowNode(*shadowNode);
+  auto props = newestCloneOfShadowNode->getProps();
+  auto viewProps = std::static_pointer_cast<const ViewProps>(props);
+  auto layoutableShadowNode =
+      traitCast<LayoutableShadowNode const *>(newestCloneOfShadowNode.get());
+  const auto &frame = layoutableShadowNode->layoutMetrics_.frame;
+
+  if (propName == "width") {
+    return std::to_string(frame.size.width);
+  } else if (propName == "height") {
+    return std::to_string(frame.size.width);
+  } else if (propName == "top") {
+    return std::to_string(frame.origin.y);
+  } else if (propName == "left") {
+    return std::to_string(frame.origin.x);
+  } else if (propName == "opacity") {
+    return std::to_string(viewProps->opacity);
+  } else if (propName == "zIndex") {
+    std::optional<int> zIndex = viewProps->zIndex;
+    if (zIndex) {
+      return std::to_string(*zIndex);
+    }
+  } else if (propName == "backgroundColor") {
+    return int_to_hex(*viewProps->backgroundColor);
+  }
+
+  return "Getting property '" + propName +
+      "' with function 'getViewProp' is not supported";
+}
+
+#endif
+
+#ifdef RCT_NEW_ARCH_ENABLED
```

Comment:
This `#endif` ends `#ifdef RCT_NEW_ARCH_ENABLED` so I think we can just remove these lines?
```suggestion
```